### PR TITLE
Feature/ade 765 create lambda proxy

### PR DESCRIPTION
**Pull request by SamMurphyDev**
Both awsScim.ts and awsScim.test.ts are hidden by default as they are large files. Will need to open to view.

### CODE REVIEW

File: src/handlers/awsScim.test.ts [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+          authorization: event.headers.authorization,
+          // @ts-ignore
+          host: new URL(process.env.PROXY_URL).hostname,
+        },
+      })
+    );
+  });
+
+  it('should respond with downstream statuses', async () => {
+    const statusesCodes = [
```
Comment (dspasojevic) [inline src/handlers/awsScim.test.ts]: Prefer that to define data outside the test case and then use a data driven test.

**Comment by dspasojevic** [comment]
@SamMurphyDev as discussed please:

1. Refactor your handler so that transforms are in a separately testable unit.
2. Change the transformers that need to fetch to accept a function that they can call to get the data.
3. Use axios-retry to back-off when a 429 response comes back.

File: src/handlers/awsScim.test.ts (hunk 1 above)
Comment (SamMurphyDev) [reply src/handlers/awsScim.test.ts]: Move a lot of the tests to data-driven tests. There are still a couple that has data inside the test case.

File: src/services/scimTransformation.ts [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+> => {
+  const [rawPath, groupId] = getTenancyAndGroupFromPath(groupPath) ?? [
+    undefined,
+  ];
+
+  if (rawPath === undefined || groupId === undefined) {
+    return undefined;
+  }
+
+  const currentGroupMembers = await fetchUsersInGroup(
```
Comment (dspasojevic) [inline src/services/scimTransformation.ts]: In `scimTransformation` can you pass this in as a function instead?

Comment (SamMurphyDev) [reply src/services/scimTransformation.ts]: Move all fetch function definitions out of the transformation logic back up to the handler

**Review by dspasojevic — APPROVED** [review]

**Pull request merged**